scripts/models/ngram_model/ngram_lm.py

```diff
@@ -0,0 +1,250 @@
+"""This module implements a ngram language model."""
+
+# import standard python modules
+from abc import ABC, abstractmethod
+import random
+from typing import Union, List, Tuple, Iterator
+import pickle
+import logging
+from pathlib import Path
+from itertools import tee
+from collections import defaultdict
+
+# import downloaded packages
+import numpy as np
+
+random.seed(1798)
+
+LOGGER = logging.getLogger(__name__)
+logging.basicConfig(level=logging.DEBUG)
+
+Ngram = Tuple[str]
+
+class BaseNgramLM(ABC):
+    """
+    An abstract class for ngram language model.
+
+    Parameters
+    ---------
+    - smooth: float, int
+        Value for smoothing the probability distributions\
+        for dealing with unknown units.
+    - pad_utterances: bool
+        Whether add or not fake tokens at the begenning and ending of the utterance.
+    - parameters: defaultdict
+        Will contain the ngrams counts.
+    """
+    def __init__(self, smooth: Union[float, int]=1e-3):
```

Comment:
Correct me if I am wrong, the indication that smooth should be int or float is purely for documentation and indicating that both types can be used?

Reply:
yes, the smooth value is in range ]0;1], so can also be 1 (an int)